lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,192 @@
+    def list_all_project_variables(self):
+        raw_variable_list = self.project.variables.list()
+        retval = []
+        if len(raw_variable_list) > 0:
+            for item in raw_variable_list:
+                retval.append(item.get_id())
+        return retval
+
+    def create_variable(self, key, value):
+        return self.project.variables.create({"key": key, "value": value})
+
+    def update_variable(self, key, value):
+        var = self.project.variables.get(key)
+        if var.value == value:
+            return False
+        var.save()
+        return True
+
+    def delete_variable(self, key):
+        return self.project.variables.delete(key)
+
+
+def native_python_main(server_url, login_token, project_name, purge_vars, var_list, state):
+
+    change = False
+    this_gitlab = gitlab_project_variables(
+        login_token=login_token, project_name=project_name, server_url=server_url)
+
+    existing_variables = this_gitlab.list_all_project_variables()
+
+    for idx in range(len(var_list)):
+        key = list(var_list[idx].keys())[0]
+        if key in existing_variables and state == 'present':
+            change = this_gitlab.update_variable(
+                key, var_list[idx][key]) or change
+            pop_index = existing_variables.index(key)
+            existing_variables.pop(pop_index)
+        elif key not in existing_variables and state == 'present':
+            this_gitlab.create_variable(key, var_list[idx][key])
+            change = True
+        elif key in existing_variables and state == 'absent':
+            this_gitlab.delete_variable(key)
+            change = True
+
+    if len(existing_variables) > 0 and purge_vars:
+        for item in existing_variables:
+            this_gitlab.delete_variable(item)
+            change = True
+
+    existing_variables = this_gitlab.list_all_project_variables()
+    return change
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(required=True, type='str'),
+            login_token=dict(required=True, type='str'),
+            name=dict(required=True, type='str'),
+            purge_vars=dict(required=False, default=False, type='bool'),
+            vars=dict(required=False, default=list(), type='list'),
+            state=dict(type='str', default="present", choices=["absent", "present"])
+        )
+    )
+
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(
+            msg="Missing required gitlab module (check docs or install with: pip install python-gitlab")
+
+    server_url = module.params['server_url']
+    login_token = module.params['login_token']
+    purge_vars = module.params['purge_vars']
+    var_list = module.params['vars']
+    project_name = module.params['name']
+    state = module.params['state']
+
+    change = native_python_main(
+        server_url, login_token, project_name, purge_vars, var_list, state)
+
+    module.exit_json(changed=change, gitlab_project_variables=None)
```

Comment:
Should be interesting to return data to the user, like variables `added`, `updated`, `removed